classes/MerchantAPI.php

```diff
@@ -0,0 +1,179 @@
+        }
+
+        return $response->getPrimaryPayment();
+    }
+
+    /**
+     * Method to trigger capture action
+     * @param $paymentId
+     * @param array $orderLines
+     * @param int $amount
+     * @return AltapayCaptureResponse
+     * @throws AltapayMerchantAPIException
+     */
+    public function captureAmount($paymentId, $orderLines = array(), $amount = 0)
+    {
+        $response = $this->api->captureReservation($paymentId, $amount, $orderLines);
+
+        if (!$response->wasSuccessful()) {
+            $xmlResponse = $this->xmlParser($response->getXml());
+            throw new Exception($this->errorMsg($xmlResponse, 'Capture'));
+        }
+
+        return $response;
+    }
+
+    /**
+     * Method to trigger refund action
+     * @param $paymentId
+     * @param array $orderLines
+     * @param int $amount
+     * @return AltapayRefundResponse
+     * @throws AltapayMerchantAPIException
+     */
+    public function refundAmount($paymentId, $orderLines = array(), $amount = 0)
+    {
+        $response = $this->api->refundCapturedReservation($paymentId, $amount, $orderLines);
+
+        if (!$response->wasSuccessful()) {
+            $xmlResponse = $this->xmlParser($response->getXml());
+            throw new Exception($this->errorMsg($xmlResponse, 'Refund'));
+        }
+
+        return $response;
+    }
+
+    /**
+     * Method to trigger release action
+     * @param $paymentId
+     * @param $transactionAction
+     * @return AltapayReleaseResponse
+     * @throws AltapayMerchantAPIException
+     */
+    public function release($paymentId, $transactionAction)
+    {
+        $response = $this->api->releaseReservation($paymentId);
+
+        if (!$response->wasSuccessful()) {
+            $xmlResponse = $this->xmlParser($response->getXml());
+            throw new Exception($this->errorMsg($xmlResponse, $transactionAction));
+        }
+
+        return $response;
+    }
+
+    /**
+     * Method for validation of merchant details
+     * @throws AltapayMerchantAPIException
+     */
+    private function validateConfiguration()
+    {
+        if (empty($this->api_url) ||
+            empty($this->api_username) ||
+            empty($this->api_password)) {
+            throw new Exception('Url, username or password missing');
+        }
+
+        $this->api = new AltapayMerchantAPI(
+            $this->api_url,
+            $this->api_username,
+            $this->api_password,
+            null
+        );
+        $response = $this->api->login();
+        if (!$response->wasSuccessful()) {
+            throw new Exception("Could not login to the Merchant API: ".
+                $response->getErrorMessage(), $response->getErrorCode());
+        }
+    }
+
+    /**
+     * @param $xml
+     * @return mixed
+     */
+    public function xmlParser($xml)
+    {
+        //get the response XML and convert into array
+        $xml = simplexml_load_string($xml);
```

Comment:
This is missing error handeling, the simples way to do so is to instanciate the object instead:
```suggestion
        $xml = new SimpleXMLElement($xml);
```